flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ShowTablesOperation.java

```diff
@@ -21,8 +21,83 @@
 /** Operation to describe a SHOW TABLES statement. */
 public class ShowTablesOperation implements ShowOperation {
 
+    private final String catalogName;
+    private final String databaseName;
+    private final boolean useLike;
+    private final boolean notLike;
+    private final String likePattern;
+    private final String preposition;
+
+    public ShowTablesOperation() {
+        this.catalogName = null;
+        this.databaseName = null;
+        this.likePattern = null;
+        this.useLike = false;
+        this.notLike = false;
+        this.preposition = null;
+    }
+
+    public ShowTablesOperation(String likePattern, boolean useLike, boolean notLike) {
+        this.catalogName = null;
+        this.databaseName = null;
+        this.likePattern = likePattern;
+        this.useLike = useLike;
+        this.notLike = notLike;
+        this.preposition = null;
+    }
+
+    public ShowTablesOperation(
+            String catalogName,
+            String databaseName,
+            String likePattern,
+            boolean useLike,
+            boolean notLike,
+            String preposition) {
+        this.catalogName = catalogName;
+        this.databaseName = databaseName;
+        this.likePattern = likePattern;
+        this.useLike = useLike;
+        this.notLike = notLike;
```

Comment:
Please add some validations such as `likePattern` shouldn't be null when `useLike` is true.

Reply:
done